### PR TITLE
Updating how date is queried for reader search

### PR DESCRIPTION
**Pull request by tejans24**
connects #1789

### CODE REVIEW

File: app/assets/stylesheets/_reviewer.scss [hunk 1]
```diff
@@ -493,7 +493,7 @@ input[type="search"].cf-search-input-with-close {
 
 [type="button"].cf-search-close-icon {
   margin-left: -20px;
-  margin-top: 6px;
+  margin-top: 2px;
```
Comment (mdbenjam) [inline app/assets/stylesheets/_reviewer.scss]: I'd remove this all together. I think it looks better with 0px of top margin

Comment (mdbenjam) [inline app/assets/stylesheets/_reviewer.scss]: Thanks for fixing this, it was bothering me... Not sure what changed to mess it up.

File: client/app/reader/search.js [hunk 2]
```diff
@@ -4,8 +4,7 @@ import { makeGetAnnotationsByDocumentId } from './selectors';
 import { doDatesMatch } from '../util/DateUtil';
 
 const metadataContainsString = (searchQuery, doc) => {
```
Comment (mdbenjam) [inline client/app/reader/search.js]: lets change the name of this method from `metadataContainsString` to `typeContainsString`

File: client/app/util/DateUtil.js [hunk 3]
```diff
@@ -25,29 +25,56 @@ const DAY_INDEX = 2;
...
-  // MM-DD-YYYY format
-  const updatedDocDateTokens = [docDateTokens[MONTH_INDEX], docDateTokens[DAY_INDEX], docDateTokens[YEAR_INDEX]];
   let searchQueryTokens = query.toLowerCase().split(DASH);
 
   // if no dashes exist in the query
   if (!_.includes(query, DASH)) {
     searchQueryTokens = query.toLowerCase().split(SLASH);
   }
 
+  // returning after removing empty strings from the tokens
```
Comment (mdbenjam) [inline client/app/util/DateUtil.js]: This comment doesn't really make sense to me. I don't see you removing empty strings here.

File: client/app/util/DateUtil.js [hunk 4]
```diff
@@ -25,29 +25,56 @@ const DAY_INDEX = 2;
...
 };
+
+const getDateTokens = (date) => {
+  let docDateTokens = date.split(DASH);
+
+  return [docDateTokens[MONTH_INDEX], docDateTokens[DAY_INDEX], docDateTokens[YEAR_INDEX]];
+};
+
+export const doDatesMatch = (date, query) => {
+  // move year to the end of the array to match
```
Comment (mdbenjam) [inline client/app/util/DateUtil.js]: I'd move this comment into `getDateTokens`

File: client/test/karma/util/DateUtil-test.js [hunk 5]
```diff
@@ -23,10 +23,12 @@ describe('DateUtil', () => {
       expect(doDatesMatch('2017-06-12', '6/')).to.be.true;
       expect(doDatesMatch('2022-02-12', '02-')).to.be.true;
       expect(doDatesMatch('2022-02-01', '2-1-2022')).to.be.true;
-      expect(doDatesMatch('2022-02-12', '2022-02-12')).to.be.false;
       expect(doDatesMatch('2022-02-12', '-02-12')).to.be.false;
       expect(doDatesMatch('2022-02-12', 'a2-12-22')).to.be.false;
       expect(doDatesMatch('2022-02-12', '02-2022-02')).to.be.false;
+      expect(doDatesMatch('2022-02-12', '2017 ')).to.be.false;
+      expect(doDatesMatch('2022-02-12', '/2022')).to.be.true;
+      expect(doDatesMatch('2022-02-12', '//2022')).to.be.true;
```
Comment (mdbenjam) [inline client/test/karma/util/DateUtil-test.js]: I don't think we should have this last test. That's not expected behavior. It's fine that it does it, but I don't think we need to test for it.

Comment (mdbenjam) [inline client/test/karma/util/DateUtil-test.js]: We should probably add some tests for searching by date.

File: client/app/reader/search.js [hunk 6]
```diff
@@ -25,11 +24,13 @@ const tagContainsString = (searchQuery, doc) =>
   }
   , false);
 
-export const searchString = (searchQuery, state) => (doc) =>
-  !searchQuery || searchQuery.split(' ').some((searchWord) => {
+export const searchString = (searchQuery, state) => (doc) => {
```
Comment (mdbenjam) [inline client/app/reader/search.js]: Any reason we made this a non-one-liner?

File: client/app/util/DateUtil.js [hunk 7]
```diff
@@ -56,14 +55,14 @@ const doesQueryMatchDateTokens = (docDateTokens, searchQueryTokens) => {
 const getDateTokens = (date) => {
   let docDateTokens = date.split(DASH);
 
+  // move year to the end of the array to match
   return [docDateTokens[MONTH_INDEX], docDateTokens[DAY_INDEX], docDateTokens[YEAR_INDEX]];
 };
 
 export const doDatesMatch = (date, query) => {
-  // move year to the end of the array to match
   // MM-DD-YYYY format
```
Comment (mdbenjam) [inline client/app/util/DateUtil.js]: I think this comment should also have been moved. It doesn't really make sense here.

File: spec/feature/reader_spec.rb [hunk 8]
```diff
@@ -662,6 +662,24 @@ def get_aria_labels(elems)
       expect(page).to have_content("Form 9")
     end
 
+    scenario "When user is searching for reciept dates" do
```
Comment (mdbenjam) [inline spec/feature/reader_spec.rb]: I think this test might be better in DecisionReviewer-test. We already have search tests there.

Comment (tejans24) [inline spec/feature/reader_spec.rb]: agreed, I'm just looking at those tests again and that makes more sense.

File: client/app/reader/search.js [hunk 9]
```diff
@@ -24,13 +24,15 @@ const tagContainsString = (searchQuery, doc) =>
   }
   , false);
 
-export const searchString = (searchQuery, state) => (doc) => {  
+export const searchString = (searchQuery, state) => (doc) => {
+  const dateMatch = doDatesMatch(doc.receivedAt, searchQuery);
```
Comment (mdbenjam) [inline client/app/reader/search.js]: nit: Why did you move this outside of the or block? The nice thing about the or block is it will only evaluate until something is true. Doesn't really matter since this is the first check.

File: client/app/util/DateUtil.js [hunk 10]
```diff
@@ -26,11 +26,13 @@ const DASH = '-';
 const SLASH = '/';
 
 const parseQueryToTokens = (query) => {
-  // date format passed in needs be in YYYY-MM-DD
-  // example: 2016-06-12
-  let searchQueryTokens = query.toLowerCase().split(DASH);
+  // date format passed in needs be in YYYY-MM-DD format
+  // For example: 2016-06-12
+  let searchQueryTokens = query ? query.trim().toLowerCase().
```
Comment (mdbenjam) [inline client/app/util/DateUtil.js]: When can query be null?
